etcdserver/api/v3rpc/watch.go

```diff
@@ -66,6 +70,15 @@ func (ws *watchServer) Watch(stream pb.Watch_WatchServer) error {
 	return sws.recvLoop()
 }
 
+func (sws *serverWatchStream) newResponseHeader() *pb.ResponseHeader {
+	return &pb.ResponseHeader{
+		ClusterId: uint64(sws.server.ClusterID()),
+		MemberId:  uint64(sws.server.ID()),
+		Revision:  sws.watchStream.Revision(),
```

Comment:
Basically, we should return the rev when the actual Txn happens for the synced watchers.